packaging/_elffile.py

```diff
@@ -0,0 +1,99 @@
+"""ELF file parser.
+
+This provides a class ``ElfFile`` that parses an ELF executable in a similar
+interface to ``ZipFile``. Only the read interface is implemented.
+
+Based on: https://gist.github.com/lyssdod/f51579ae8d93c8657a5564aefc2ffbca
+ELF header: https://refspecs.linuxfoundation.org/elf/gabi4+/ch4.eheader.html
+"""
+
+import enum
+import os
+import struct
+from typing import IO, Optional, Tuple
+
+
+class ElfInvalid(ValueError):
```

Comment:
Keep acronyms capitalized instead of faking CapWords (it's a PEP 8 thing).

```suggestion
class ELFInvalid(ValueError):
```